advanced/Scripts/update.sh

```diff
@@ -56,14 +58,26 @@ GitCheckUpdateAvail() {
         # need @{0}
         LOCAL="$(git rev-parse "@{0}")"
 
-        # The suffix @{upstream} to a branchname
-        # (short form <branchname>@{u}) refers
-        # to the branch that the branch specified
-        # by branchname is set to build on top of#
-        # (configured with branch.<name>.remote and
-        # branch.<name>.merge). A missing branchname
-        # defaults to the current one.
-        REMOTE="$(git rev-parse "@{upstream}")"
+        # check if the local branch ref is a branch or a tag on remote repo
+        if git show-ref -q --verify "refs/remotes/origin/$curBranch" 2>/dev/null; then
+            ref_is_tag=false
+        elif git show-ref -q --verify "refs/tags/$curBranch" 2>/dev/null; then
+            ref_is_tag=true
+        fi
```

Comment:
I see no `else` here, what happens if neither of the two are true? `ref_is_tag` seems to stay random in this case (empty, likely). Can this happen? (I'm thinking about users doing local changes and committing them so the head is not pointing to anything on `origin` and also isn't a tag)